### PR TITLE
revise research folder readme

### LINKED ISSUE #621: documentation updates for 0.2.0
**Describe the solution you'd like**
this ticket tracks the documentation updates for the 0.2.0 release (early July)
- [x] double check the examples and research code
- [x] update highlights (#536 #540)
- [x] update change log (may revisit #234)
- [x] revise the contributing guide to reflect the changes of local tests https://github.com/Project-MONAI/MONAI/pull/496 https://github.com/Project-MONAI/MONAI/pull/553
- [x] revise the contributing guide to reflect the changes of minimal CI tests https://github.com/Project-MONAI/MONAI/pull/595
~- [ ] update website and double-check all the URLs https://github.com/Project-MONAI/project-monai.github.io~ (will address this in the website source code repo)
- [x] #391 
- [x] change to use `pip install monai==0.2.0` for some examples/demos
- [x] group transforms into subsections in https://github.com/Project-MONAI/MONAI/blob/master/docs/source/transforms.rst

### PR DESCRIPTION
**Pull request by wyli**
part of  #621 
closes #621

### Description
pin the research folder to use monai v0.2

### Status
**Ready**

### Types of changes

- [x] Docstrings/Documentation updated

### CODE REVIEW

File: research/coplenet-pneumonia-lesion-segmentation/README.md [hunk 1]
```diff
@@ -21,7 +21,14 @@ This research prototype is adapted from:
 
 To run the inference demo:
 
-- install the latest version of MONAI: `git clone https://github.com/Project-MONAI/MONAI` and `pip install -e .`.
+- Download and switch to MONAI 0.2.0 source code:
+```bash
+git clone https://github.com/Project-MONAI/MONAI
+cd MONAI
+git checkout 0.2.0
```
Comment (Nic-Ma) [inline research/coplenet-pneumonia-lesion-segmentation/README.md:28]: 1. Is it `releases/0.2.0` or `0.2.0`?
2. I think maybe it's easier to download 0.2.0 branch directly?
`git clone https://github.com/Project-MONAI/MONAI --branch releases/0.2.0`

Thanks.

File: research/lamp-automated-model-parallelism/README.md [hunk 2]
```diff
@@ -17,21 +17,28 @@ MICCAI 2020 (Early Accept, paper link: https://arxiv.org/abs/2006.12575)
 ## To run the demo:
 
 ### Prerequisites
-- install the latest version of MONAI: `git clone https://github.com/Project-MONAI/MONAI` and `pip install -e .`
+- Download and switch to MONAI 0.2.0 source code:
+```bash
+git clone https://github.com/Project-MONAI/MONAI
+cd MONAI
+git checkout 0.2.0
+pip install -e .  # install from the source code
```
Comment (Nic-Ma) [inline research/lamp-automated-model-parallelism/README.md:25]: Similar to the coplenet, maybe simplify to below?
```
pip install -e '.[torchgpipe]`
```

Thanks.

File: research/coplenet-pneumonia-lesion-segmentation/README.md (hunk 1 above)
Comment (wyli) [reply research/coplenet-pneumonia-lesion-segmentation/README.md:28]: I think the current way is more clear, and I've checked it works fine

File: research/lamp-automated-model-parallelism/README.md (hunk 2 above)
Comment (wyli) [reply research/lamp-automated-model-parallelism/README.md:25]: why do we add `torchgpipe`? there're packages like matplotlib, Jupiter notebook in the examples, we don't introduce them as (optional) deps

Comment (Nic-Ma) [reply research/lamp-automated-model-parallelism/README.md:25]: Oh, I see, so this `torchgpipe` is not a common lib also for other works, right?

**Pull request merged**